plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -245,6 +256,32 @@ public class SqlServerClient
         return FULL_PUSHDOWN.apply(session, simplifiedDomain);
     };
 
+    // Dates prior to the Gregorian calendar switch in 1582 can cause incorrect results when pushed down,
+    // so we disable predicate push down when the domain contains values prior to 1583
+    private static final Instant DISABLE_DATETIMEOFFSET_PUSHDOWN_IF_BEFORE = Instant.parse("1583-01-01T00:00:00Z");
+    private static final LongTimestampWithTimeZone LONG_DATETIMEOFFSET_DISABLE_VALUE =
+            LongTimestampWithTimeZone.fromEpochSecondsAndFraction(
+                    DISABLE_DATETIMEOFFSET_PUSHDOWN_IF_BEFORE.getEpochSecond(),
+                    (long) DISABLE_DATETIMEOFFSET_PUSHDOWN_IF_BEFORE.getNano() * PICOSECONDS_PER_NANOSECOND,
+                    UTC_KEY);
+    private static final long SHORT_DATETIMEOFFSET_DISABLE_VALUE = DISABLE_DATETIMEOFFSET_PUSHDOWN_IF_BEFORE.toEpochMilli();
+
+    private static final PredicatePushdownController SQLSERVER_DATE_TIME_PUSHDOWN = (session, domain) -> {
+        Domain simplifiedDomain = domain.simplify(getDomainCompactionThreshold(session));
+        for (Range range : simplifiedDomain.getValues().getRanges().getOrderedRanges()) {
+            Range disableRange = range.getType().getJavaType().equals(LongTimestampWithTimeZone.class)
```

Comment:
Nit: should we keep these two ranges as constants, instead of `LONG_DATETIMEOFFSET_DISABLE_VALUE` and `SHORT_DATETIMEOFFSET_DISABLE_VALUE`?

Reply:
@nineinchnick I had tried that, but the issue with having the `Range` as constants is you need to create them like below, but these class types cannot be imported because they aren't `public` in `io.trino.spi.type`. Please let me know if there is another way to create them.

```
Range.lessThan(LongTimestampWithTimeZoneType.class, LONG_DATETIMEOFFSET_DISABLE_VALUE)
Range.lessThan(ShortTimestampWithTimeZoneType.class, SHORT_DATETIMEOFFSET_DISABLE_VALUE);
```